OpenSim/Common/MarkerData.cpp

```diff
@@ -486,8 +486,9 @@ void MarkerData::readStoFile(const string& aFileName)
     // If the file was written by STOFileAdapter, make the file readable by
     // Storage. Calls below have no effect otherwise.
     std::string tmpFileName{"tmp.sto"};
-    revertToVersionNumber1(aFileName, tmpFileName);
-    addNumRowsNumColumns(tmpFileName, aFileName);
+    bool versionChanged{revertToVersionNumber1(aFileName, tmpFileName)};
+    if(versionChanged)
+        addNumRowsNumColumns(tmpFileName, aFileName);
```

Comment:
The function `revertToVersionNumber1()` was written earlier to get some of the tests working, which use STO files that don't need adding number of rows and columns to header. Tests currently using this function -- `testControllerExample`, `testCustomActuatorExample`, `testExampleMain` and so on. 
This PR adds the function `addNumRowsColumns()` as a compatibility (between `STOFileAdapter` and `MarkerData`) fix.
So both these functions need to exist, unless we change the STO files used by the tests I mention above. I can make the change if you like.